File: task-03/Ruby/Subtask3.rb
#  Ask the user to enter a number
puts "Enter a number:"
n = gets.to_i  

# First half of the diamond
for i in 0...n
  spaces = n - i - 1
  stars = 2 * i + 1
  puts " " * spaces + "*" * stars
end
# Second half of the diamond
for i in (n-2).downto(0)
  spaces = n - i - 1
  stars = 2 * i + 1
  puts " " * spaces + "*" * stars
end

File: task-03/C/Subtask3.c
#include <stdio.h>

int main() {
    int n;

    printf("Enter the number of rows: ");
    scanf("%d", &n);

    // First half of the diamond
    for (int i = 0; i < n; i++) {
        // Loop to print spaces
        for (int j = 0; j < n - i - 1; j++) {
            printf(" ");
        }
        // Loop to print asterisks
        for (int j = 0; j < 2 * i + 1; j++) {
            printf("*");
        }
        printf("\n"); // Move to the next line
    }

    // Second half of the diamond
    for (int i = n - 2; i >= 0; i--) {
        // Loop to print spaces
        for (int j = 0; j < n - i - 1; j++) {
            printf(" ");
        }
        // Loop to print asterisks
        for (int j = 0; j < 2 * i + 1; j++) {
            printf("*");
        }
        printf("\n"); // Move to the next line
    }

    return 0;
}

